server/routes/user.js

```diff
@@ -1,31 +1,111 @@
 const express = require("express");
-const { check, validationResult } = require("express-validator");
+const { check, body, validationResult } = require("express-validator");
 const router = express.Router();
+const bcrypt = require("bcryptjs");
 
 const { User } = require("../database");
 
 router.post(
-  "/register",
+  "/signup",
   [
     check("name", "Name is required")
       .not()
       .isEmpty(),
-    check("email", "Please use a valid email address").isEmail()
+    check("email", "Please use a valid email address").isEmail(),
+    check("password", "Choose a password at least 6 characters long").isLength({
+      min: 6
+    }),
+    body("confirmPassword").custom((value, { req }) => {
+      if (value !== req.body.password) {
+        throw new Error("Password confirmation does not match password");
+      }
+      return true;
+    })
   ],
   async (req, res) => {
     const errors = validationResult(req);
-    if (errors.length) {
+    if (errors.length > 0) {
       return res.status(400).json({ errors: errors.array() });
     }
-    const { name, email } = req.body;
+
+    const { name, email, password } = req.body;
+
     try {
-      const user = new User({ name, email });
-      await user.save();
-      res.send(user);
+      //Check for an existing user
+      var user = await User.findOne({ email: email });
```

Comment:
Mostly a style thing but you can use `{ email }` here and `{ email, name, password }`.

Reply:
Not quite sure what you mean, sorry if I'm being dense. Can you explain a bit more?